packages/deployments/contracts/contracts/lib/Connext/ConnextLogic.sol

```diff
@@ -948,10 +1161,42 @@ library ConnextLogic {
         _adoptedToLocalPools,
         _args.tokenRegistry,
         _args.executeArgs.local,
-        toSwap
+        vars.fastTransferAmount
       );
   }
 
+  /**
+   * @notice Uses Aave Portals to provide fast liquidity
+   */
+  function _executePortalTransfer(
+    bytes32 _transferId,
+    uint256 _fastTransferAmount,
+    ExecuteLibArgs calldata _args,
+    mapping(bytes32 => IStableSwap) storage _adoptedToLocalPools,
+    mapping(bytes32 => address) storage _canonicalToAdopted,
+    mapping(bytes32 => uint256) storage _aavePortalsTransfers
+  ) public returns (uint256, address) {
+    // Calculate local to adopted swap output if needed
+    (uint256 userAmount, address adopted) = AssetLogic.calculateSwapFromLocalAssetIfNeeded(
+      _canonicalToAdopted,
+      _adoptedToLocalPools,
+      _args.tokenRegistry,
+      _args.executeArgs.local,
+      _fastTransferAmount
+    );
+
+    IAavePool(_args.aavePool).mintUnbacked(adopted, userAmount, address(this), AAVE_REFERRAL_CODE);
+
+    // Improvement: Instead of withdrawing to address(this), withdraw directly to the user or executor to save 1 transfer
```

Comment:
🤤

Reply:
I thought about adding this, but it introduces some complexity in the code, and also it is only an improvement if the asset is not a native asset that needs to be unwrapped (so it also requires additional checks). So I won't include this for now